File: Python-Basics/loops/vacation.py
vacation_price = float(input())
current_balance = float(input())

times_spending = 0
days = 0


while current_balance < vacation_price:
    action = input()
    action_sum = float(input())
    days += 1
    if action == 'spend':
        if action_sum > current_balance:
            current_balance = 0
        else:
            current_balance -= action_sum
        times_spending += 1
    else:
        current_balance += action_sum
        times_spending = 0

    if times_spending == 5:
        print(f"You can't save the money")
        print(days)
        break


if times_spending != 5:
    print(f'You saved the money for {days} days.')

File: CPP-Basics/Vacation.cpp
#include <iostream>
#include <string>

using namespace std;

int main () {

    double neededMoney;
    cin >> neededMoney;

    double balance;
    cin >> balance;

    int dayCount = 0;
    int spendCount = 0;

    do {

        string action;
        cin >> action;

        double amount;
        cin >> amount;

        if(action == "save") {

            balance += amount;
            spendCount = 0;

        } else if (action == "spend") {

            balance -= amount;
            spendCount ++;
        }

        if (balance < 0) {

            balance = 0;
        }

        dayCount++;
    }

    while(balance < neededMoney && spendCount < 5);

    if(balance >= neededMoney) {

        cout << "You saved the money for " << dayCount << " days." << endl;
    }

    if(spendCount == 5) {

        cout << "You can't save the money." << endl << dayCount << endl;
    }

    return 0;
}
